src/traits.d

```diff
@@ -112,48 +112,29 @@ static this()
 /**
  * get an array of size_t values that indicate possible pointer words in memory
  *  if interpreted as the type given as argument
- * the first array element is the size of the type for independent interpretation
- *  of the array
- * following elements bits represent one word (4/8 bytes depending on the target
- *  architecture). If set the corresponding memory might contain a pointer/reference.
- *
- *  [T.sizeof, pointerbit0-31/63, pointerbit32/64-63/128, ...]
+ * returns the size of the type in bytes, d_uns64.max on error
  */
-extern (C++) Expression pointerBitmap(TraitsExp e)
+extern (C++) d_uns64 getTypePointerBitmap(Loc loc, Type t, Array!(d_uns64)* data)
 {
-    if (!e.args || e.args.dim != 1)
-    {
-        error(e.loc, "a single type expected for trait pointerBitmap");
-        return new ErrorExp();
-    }
-
-    Type t = getType((*e.args)[0]);
-    if (!t)
-    {
-        error(e.loc, "%s is not a type", (*e.args)[0].toChars());
-        return new ErrorExp();
-    }
-
     d_uns64 sz;
     if (t.ty == Tclass && !(cast(TypeClass)t).sym.isInterfaceDeclaration())
-        sz = (cast(TypeClass)t).sym.AggregateDeclaration.size(e.loc);
+        sz = (cast(TypeClass)t).sym.AggregateDeclaration.size(loc);
     else
-        sz = t.size(e.loc);
+        sz = t.size(loc);
     if (sz == SIZE_INVALID)
-        return new ErrorExp();
+        return d_uns64.max;
```

Comment:
There is no functional change here. getTypePointerBitmap() has just been extracted from pointerBitmap() to be callable from elsewhere. Having to deal with the Expression return value in write_instance_pointers() would be wasteful.